src/operator/mkldnn/mkldnn_memory-inl.h

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file mkldnn_memory-inl.h 
+ * \brief
+ * \author Chen, Xiaoming 
+*/
+
+#ifndef MXNET_OPERATOR_MKLDNN_MKLDNN_MEMORY_INL_H_
+#define MXNET_OPERATOR_MKLDNN_MKLDNN_MEMORY_INL_H_
+
+#include <dmlc/logging.h>
+#include "./mkldnn_cppwrapper.h"
+namespace mxnet {
+#if MXNET_USE_MKLDNN == 1
+template <typename DType>
+struct MKLMemoryDescriptorBase {
```

Comment:
Does this hold actual input/output buffers or just descriptors? What's the memory used for?
If it's actual data, consider using mxnet's tempspace so that it can be shared to save memory.

This is probably a hard fix, we can do it in the next release